smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/ImportDeclarations.kt

```diff
@@ -8,8 +8,22 @@ package software.amazon.smithy.swift.codegen
 class ImportDeclarations {
     private val imports = mutableSetOf<ImportStatement>()
 
-    fun addImport(packageName: String, isTestable: Boolean = false) {
-        imports.add(ImportStatement(packageName, isTestable))
+    fun addImport(packageName: String, isTestable: Boolean = false, internalSPIName: String? = null) {
+        val existingImport = imports.find { it.packageName == packageName }
```

Comment:
Do we need to search for existing imports by package name _and_ internal SPI name?

i.e. you can `import ABC` and `import @_spi(DEF) ABC` in the same source file, right?

Reply:
Nope. There'd be no reason to `import ABC` and `import @_spi(DEF) ABC` in the same source file.

`import @_spi(DEF) ABC` will import all of `ABC` plus it'll import anything that is marked with `@_spi(DEF)`.

Reply:
What if we imported:
```
import @_spi(DEF) ABC
import @_spi(GHI) ABC
```
These are two distinct imports but this code would wipe out one of these two, no?

Reply:
Correct, this currently doesn't support adding multiple SPI internal names. I can add support for that if its something we want